src/server.rs

```diff
@@ -645,14 +649,22 @@ impl Seabird for Arc<Server> {
             };
 
             loop {
-                let event = events.recv().await.map_err(|err| {
-                    Status::internal(format!("failed to read internal event: {:?}", err))
-                })?;
-
-                sender
-                    .send(Ok(event))
-                    .await
-                    .map_err(|err| Status::internal(format!("failed to send event: {}", err)))?;
+                match select(events.next(), &mut notifier).await {
+                    Either::Left((Some(Ok(event)), _)) => {
+                        sender.send(Ok(event)).await.map_err(|err| {
+                            Status::internal(format!("failed to send event: {}", err))
+                        })
+                    }
+                    Either::Left((Some(Err(err)), _)) => {
+                        Err(Status::internal(format!("failed to read internal event: {:?}", err)))
+                    }
+                    Either::Left((None, _)) => {
+                        Err(Status::internal(format!("this means something")))
```

Comment:
🤔 Is there anything more descriptive that we can do here?

Reply:
Did some digging - this means that the input stream ended, so I'm adding a more descriptive error.